### ISSUE TITLE
PlayWright process fails during robot tests

### ISSUE THREAD

**Issue by emakaay** [MarketSquare/robotframework-browser#517]
I'm in the progress to convert existing robot framework test using SeleniumLibrary to robotframework-browser. I started with version 1.4.0, and now i'm on 2.2.0 (thanks for fixing the missed upload to pypi.org)

All the time I have a small problem with one of the projects that i'm converting. When a test is run, some unexpected situation occurs, and robot shows the following error messages:
- Keyword 'Take ScreenShot' could not be run on failure:
failed to connect to all addresses
Stream removed

After this specific test, all other remaining test fail with:
ConnectionError: PlayWright process has been terminated with code 1

I don't know what is causes this problem, but here is the relevant piece of debug information from the playwright log
[debug-log-playwright.txt](https://github.com/MarketSquare/robotframework-browser/files/5579815/debug-log-playwright.txt)

maybe equal to this problem ? (https://github.com/MarketSquare/robotframework-browser/issues/463)

**Comment by mkorpela** [comment]
Seems to be related to dialogs:
```
waiting for element to be visible, enabled and not moving2020-11-22T16:10:40.789Z pw:api   element is visible, enabled and does not move2020-11-22T16:10:40.789Z pw:api   scrolling into view if needed2020-11-22T16
          triggerUncaughtException(err, true /* fromPromise */);
          ^

dialog.accept: Cannot accept dialog which is already handled!
    at Connection.sendMessageToServer (c:\python37\lib\site-packages\Browser\wrapper\node_modules\playwright\lib\client\connection.js:69:15)
    at Proxy.<anonymous> (c:\python37\lib\site-packages\Browser\wrapper\node_modules\playwright\lib\client\channelOwner.js:54:53)
    at c:\python37\lib\site-packages\Browser\wrapper\node_modules\playwright\lib\client\dialog.js:38:33
    at Dialog._wrapApiCall (c:\python37\lib\site-packages\Browser\wrapper\node_modules\playwright\lib\client\channelOwner.js:80:34)
    at Dialog.accept (c:\python37\lib\site-packages\Browser\wrapper\node_modules\playwright\lib\client\dialog.js:37:21)
    at Object.<anonymous> (c:\python37\lib\site-packages\Browser\wrapper\index.js:12683:81)
    at step (c:\python37\lib\site-packages\Browser\wrapper\index.js:12410:23)
    at Object.next (c:\python37\lib\site-packages\Browser\wrapper\index.js:12391:53)
    at c:\python37\lib\site-packages\Browser\wrapper\index.js:12385:71
    at new Promise (<anonymous>)
```

**Comment by emakaay** [comment]
Hello Mikko,

You're right. It's related to a dialog.

I don't know how it is implemented but this what i see happening (outside
robot)
1) User select a file
2) when file is selected, a import button is clicked
3) while the file is being processed a window/dialog (without buttons) is
shown
4) at the end a dialog box (with ok button) is shown to indicate that file
is processed completely

In the robot test I only have 1 "Handle Future Dialog    accept"  line

my guess is that the window/dialog as described at point 3 is throwing off
the library. The handle future dialog was meant for the dialog at point 4.
That's also the only dialog that needs accepting..

I hope this helps in troubleshooting. If you need extra information please
let me know. If I can i will try to help

**Comment by emakaay** [comment]
Hello Miko,

I'm working on converting another project to robotframework-browser.
I'm having the same problem with the dialog.

I have some tests were a single dialog is tested, they went fine.
I noticed that some tests check on multiple dialog in a single test. It fails at the second dialog. On further investigation this was also the case in the first project..... Maybe that helps ?

**Comment by emakaay** [comment]
Tried 2 other projects to convert to robotframework-browser.

Again I'm having problems with the dialogs. See attached file.
I would like to start using robotframework-browser instead of seleniumLibrary, but 2 issues are against it:
1) Handling of the dialogs. Especially when more then one dialog needs to be handled
2) Checking the text shown in the dialog (https://github.com/MarketSquare/robotframework-browser/issues/440)

Hope these problems will be fixed. Nothing else is holding me back 

[playwright-log-obfuscated.txt](https://github.com/MarketSquare/robotframework-browser/files/5644603/playwright-log-obfuscated.txt)

**Comment by emakaay** [comment]
Hello Mikko,

Today I downgraded nodejs from version 15.x to 14.15.1 (latest LTS).
And the problem with crashing went away.

I use chocolatey to download the latest software, and retrieved a new version of nodejs every 2 week or so. I also upgraded pip library robotframework-browser frequently. And every time I perform a rfbrowser init commando. To get the required/updated npm modules.

I don't know why it works now. The reason for downgrading was not because I expected a problem with nodejs.

But for me, this issue can be closed ( I re-tested 5 projects, which had problems with the dialogs, they all went fine now)
